lib/Index/IndexSymbol.cpp

```diff
@@ -254,6 +254,30 @@ SymbolInfo index::getSymbolInfoForDecl(const Decl *D) {
 
   if (isLocalSymbol(D)) {
     info.Properties |= SymbolProperty::Local;
+  } else if (auto *VD = dyn_cast<ValueDecl>(D)) {
+    switch (VD->getEffectiveAccess()) {
```

Comment:
Access level was added to the index request (not the index itself) in https://github.com/swiftlang/swift/pull/31942 and there was a long discussion there about whether it should be effective access or `getFormalAccessScope`, with the conclusion being `getFormalAccessScope`.

One notable case there was:
```
private enum E {
    case a, b, c
}
extension E {
    private func f() { print("Hello") }
}
```

Where apparently `getEffectiveAccess` gives `fileprivate` for `f` (which it isn't).